### PR TITLE
[MINT-3199] Custom Category Form Modal

### PR DESCRIPTION
**Pull request by garyjzhao**
# MINT-3199

## Description

* Added `CategoryForm` component and test and copy
* Added `CreateMTOCategory` and `GetMTOCategories` query and its types
* Updated css and components to the generic `Modal` component
* Added temporary button section for the Custom Category Modal in `ModelToOperations/Home/index.tsx`

## How to test this change

1.  navigate to MTO Home
2. Click on `Add Custom Category` button
3. Verify that the design matches figma
4. Ensure that creation of primary category works
5. After creating a primary category, create another custom category and in the dropdown, check to see if previous newly created category is in the dropdown

## PR Author Checklist

- [X] I have provided a detailed description of the changes in this PR.
- [x] I have provided clear instructions on how to test the changes in this PR.
- [x] I have updated tests or written new tests as appropriate in this PR.
- [ ] Updated the [Postman Collection](../MINT.postman_collection.json) if necessary.

## PR Reviewer Guidelines

- It's best to pull the branch locally and test it, rather than just looking at the code online!
- When approving a PR, provide a reason _why_ you're approving it
  - e.g. "Approving because I tested it locally and all functionality works as expected"
  - e.g. "Approving because the change is simple and matches the Figma design"
- Don't be afraid to leave comments or ask questions, especially if you don't understand why something was done! (This is often a great time to suggest code comments or documentation updates)
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.

### CODE REVIEW

File: src/features/ModelPlan/ModelToOperations/_components/OptionPanel/index.tsx [hunk 1]
```diff
@@ -101,7 +101,11 @@ const MTOOptionsPanel = () => {
 
       <Grid row gap={2} className="margin-top-4">
         {mtoOptions.map(option => (
-          <MTOOptionsCard mtoType={option} className="margin-top-2" />
+          <MTOOptionsCard
+            key={option}
+            mtoType={option}
+            className="margin-top-2"
+          />
```
Comment (garyjzhao) [inline src/features/ModelPlan/ModelToOperations/_components/OptionPanel/index.tsx:108]: Addressing console warning about key. not directly related to this PR

File: src/i18n/en-US/modelPlan/modelToOperations.ts [hunk 2]
```diff
@@ -42,6 +42,27 @@ export const modelToOperationsMisc: Record<string, any> = {
...
+      categoryTitle: {
+        label: 'New category title <s>*</s>'
+      },
+      sortOptions: {
+        default: '- Select - ',
+        none: 'None (this is a primary category)'
+      }
+    },
+    addButton: 'Add {{type}}',
+    cancel: 'Cancel'
```
Comment (patrickseguraoddball) [inline src/i18n/en-US/modelPlan/modelToOperations.ts:65]: For these form questions, we want to handle them to be used in change history.  This includes adding them to `types/translation.ts`.   I don't think we have anything set up yet, but the tables will need to be created and also added to the export util.  I can pair on this if you need!

**Review by patrickseguraoddball — CHANGES_REQUESTED** [review]
The mobile view is a bit off
Screenshot 2024-11-19 at 10 11 32 AM

If I initially select a primary category, then revert that to unselected, I can still submit the form, and the mutation errors out
Screenshot 2024-11-19 at 10 13 23 AM

I think we want to handle errors.  There is a mockuo in another Figma, https://www.figma.com/design/BcI7pNwfkmwaDD2EGGTsYC/IT-Lead?node-id=4704-16159&node-type=frame&t=N4Epb7NRxhceenq9-0
Might want to double check with UX

**Review by patrickseguraoddball — APPROVED** [review]
Screenshot 2024-11-21 at 2 05 04 PM


There's still this issue with the button being enabled when no category is selected.  It's not really a huge issue since the mutation will fail.  I'll approve anyway and seems like an edge case

**Pull request merged**